movieApp.js

```diff
@@ -1,13 +1,56 @@
 import { moviesDatabase } from "./movieStorage.js";
 
+// Event listeners
 document.addEventListener('DOMContentLoaded', () => {
-  const mobileToggle = document.querySelector('.mobile-toggle');
+  setupMobileToggle();
+  setupMobileSearch();
+  setupDesktopSearch();
+});
+
+// Mobile toggle
+function setupMobileToggle() {
+  const mobileToggle = document.getElementById('mobile-toggle');
   const mobileMenu = document.getElementById('mobile-menu');
 
   mobileToggle.addEventListener('click', () => {
-      mobileMenu.classList.toggle('active');
+    mobileMenu.classList.toggle('active');
+  });
+
+  document.addEventListener('click', (event) => {
+    if (!mobileMenu.contains(event.target) && !mobileToggle.contains(event.target)) {
+      mobileMenu.classList.remove('active');
+    }
+  });
+}
+
+// Mobile search
+function setupMobileSearch() {
+  const searchInputMobile = document.getElementById('search-input-mobile');
+  const mobileMenu = document.getElementById('mobile-menu');
+
+  searchInputMobile.addEventListener('input', () => {
```

Comment:
You could have a shared event handler with the desktop search by using `event.target` like this:
```js
function onSearch(event) {
  const keyword = event.target.value;
  const filteredMovies = searchMovies(keyword);
  showMovies(filteredMovies);
}
searchInputMobile.addEventListener('input', onSearch);
// ...
searchInput.addEventListener('input', onSearch);
```